### PR TITLE
Include ConnectionError for retries during EC2 instance metadata calls

### PR DESCRIPTION
**Pull request by rkenmi**

### CODE REVIEW

File: deltacat/aws/clients.py [hunk 1]
```diff
@@ -46,9 +46,9 @@ class RetryIfRetryableHTTPStatusCode(retry_if_exception):
 
     def __init__(self):
         def is_retryable_error(exception):
-            return (
+            return isinstance(exception, requests.exceptions.ConnectionError) or (
                 isinstance(exception, requests.exceptions.HTTPError)
-                and exception.response.status_code in RETRYABLE_HTTP_STATUS_CODES
+                and exception.response.status_code in RETRYABLE_HTTP_STATUS_CODES,
```
Comment (raghumdani) [inline deltacat/aws/clients.py:51]: why an additional `,` at the end? wouldn't that make it a tuple?

Comment (rkenmi) [reply deltacat/aws/clients.py]: Unintentional change, fixed.

File: deltacat/aws/clients.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ class RetryIfRetryableHTTPStatusCode(retry_if_exception):
 
     def __init__(self):
         def is_retryable_error(exception):
-            return (
+            return isinstance(exception, requests.exceptions.ConnectionError) or (
```
Comment (raghumdani) [inline deltacat/aws/clients.py:49]: Are you sure it captures all errors we saw in production?

**Review by raghumdani — APPROVED** [review]
LGTM! Please double check all errors are captured.

File: deltacat/aws/clients.py (hunk 2 above)
Comment (rkenmi) [reply deltacat/aws/clients.py:49]: We were encountering`HttpError` in production, so it is a safe bet for the time being. Adding `ConnectionError` is more of a precautionary step after looking at the request [docs](https://requests.readthedocs.io/en/latest/user/quickstart/?highlight=exceptions#errors-and-exceptions), but going forward, we can incrementally add new exceptions here as needed.

**Pull request merged**